src/BuiltInTools/dotnet-watch/README.md

```diff
@@ -91,6 +91,91 @@ Example:
+
+```json
+{
+  "type": "UpdateStaticFile",
+  "path": "wwwroot/css/site.css"
+}
+```
+
+The implementation for the static asset reload is heavily inspired by @RickStrah's Live Reload middleware located at https://github.com/RickStrahl/Westwind.AspnetCore.LiveReload
+
+Scoped css files (.razor.css and .cshtml.css) are a special case of static web assets. They need to be processed (currently requires running MSBuild targets) to be updated. Once updated, the mechanics of updating are identical to other static assets.
+
+#### Compiled files (.cs, .razor, .cshtml)
+
+Updating a compiled file requires two requirements -
+
+1. the compiler needs to able to produce a binary patch from the source changes
+1. The app needs to be able to apply the binary patch.
+
+The former has been available as a compiler feature to support debugger / IDE's Edit and Continue capabilities. Starting in 6.0, the runtime (.NET Core and Mono) [expose APIs](https://github.com/dotnet/runtime/blob/f15722c2c25ce945f2f1c7673ff3f4fbdb244feb/src/mono/System.Private.CoreLib/src/System/Reflection/Metadata/AssemblyExtensions.cs#L28) to patch a running app. The combination of the two makes it possible to hot reload changes to files that are compiled as part of the application. The first bullet point is easy for files that are already compiled as part of the app e.g. .cs or .vb files. For any other file types, source generators offer a way to participate in the compilation process. This is the route taken by Razor Source Generator for .razor and .cshtml files.
+
+To enable this, dotnet-watch hosts a Roslyn workspace that contains the project and all of it's references. When a change to .cs, .razor, or .cshtml file is found, it updates the workspace and uses it's APIs to produce a patch. Not all source changes can however produce a patch. These are called rude-edits (since it rudely interrupts your workflow) and in this case, watch falls back to doing a regular build to update and restart the app. If a patch is successfuly produced, watch now has the task of sending these patches to the app. Since different app models have different constraints, this implementation is app-model specific:
+
+* For .NET Core apps, dotnet-watch hosts a NamedPipe server. It once again uses a startup hook to inject code (`Microsoft.Extensions.AspNetCoreDeltaApplier`) in to the app that creates a client to listen to this NamedPipe. Each successful hot reload results in a binary payload with the following format:
```

Comment:
What is this dll name?

Reply:
Microsoft.Extensions.AspNetCoreDeltaApplier?

Reply:
Why is that the name of the package/dll/component?

Reply:
Ahh, as in why isn't in more generic? It probably could be, but there's ASP.NET Core specific behavior baked in to it. It's likely we'll have to refactor this to support as part of trying to reuse some of this code to support Ctrl+F5 in the IDE.

Reply:
Theni t should be Microsoft.AspNetCore.*